Spigot-Server-Patches/0009-Timings-v2.patch

```diff
@@ -428,8 +432,55 @@ index df07b2b889..f969c036f3 100644
  
      }
  
+diff --git a/src/main/java/net/minecraft/server/CustomFunction.java b/src/main/java/net/minecraft/server/CustomFunction.java
+index b7cdc495..8b9b076d 100644
+--- a/src/main/java/net/minecraft/server/CustomFunction.java
++++ b/src/main/java/net/minecraft/server/CustomFunction.java
+@@ -13,12 +13,22 @@ public class CustomFunction {
+ 
+     private final CustomFunction.c[] a;
+     private final MinecraftKey b;
++    // Paper start
++    public co.aikar.timings.Timing timing;
++    public co.aikar.timings.Timing getTiming() {
++        if (timing == null) {
++            timing = co.aikar.timings.MinecraftTimings.getCommandFunctionTiming(this);
++        }
++        return timing;
++    }
++    // Paper end
+ 
+     public CustomFunction(MinecraftKey minecraftkey, CustomFunction.c[] acustomfunction_c) {
+         this.b = minecraftkey;
+         this.a = acustomfunction_c;
+     }
+ 
++    public MinecraftKey getMinecraftKey() { return this.a(); } // Paper - OBFHELPER
+     public MinecraftKey a() {
+         return this.b;
+     }
+diff --git a/src/main/java/net/minecraft/server/CustomFunctionData.java b/src/main/java/net/minecraft/server/CustomFunctionData.java
+index 4d7e1517..1958149b 100644
+--- a/src/main/java/net/minecraft/server/CustomFunctionData.java
++++ b/src/main/java/net/minecraft/server/CustomFunctionData.java
+@@ -110,6 +110,7 @@ public class CustomFunctionData implements ITickable, IResourcePackListener {
+             return 0;
+         } else {
+             try {
++                customfunction.getTiming().startTiming(); // Paper
```

Comment:
merge this into the try as
`try (co.aikar.timing.Timing timing = customFunction.getTiming().startTiming()) {`

then remove the .stopTiming